commit aea8cc86280697c2d298ad0f8395d13f91917f75
Author: Randy Fay <randy@randyfay.com>
Date:   Wed Sep 16 17:47:51 2015 -0600

    Add args to drush command

diff --git a/mailaview.drush.inc b/mailaview.drush.inc
index 3b7db86..5071855 100644
--- a/mailaview.drush.inc
+++ b/mailaview.drush.inc
@@ -24,6 +24,7 @@ function mailaview_drush_command() {
     'options' => array(
       'view' => 'the name of the view',
       'display' => 'the id of the display',
+      'args' => 'comma-separated list of arguments',
       'to' => 'a comma separated list of email addresses to mail',
       'subject' => '(optional) the email subject to use, otherwise the view title is used',
     ),
@@ -43,7 +44,6 @@ function mailaview_drush_mailaview() {
   $display_id = drush_get_option('display');
   $to = drush_get_option('to');
   $emails = explode(",", $to);
-  // @todo Implement 'args' drush option.
   $args = drush_get_option('args', array());
   $subject = drush_get_option('subject', NULL);
   // Check if there is such view.
