diff --git a/docroot/sites/all/modules/contrib/imagecache_profiles/imagecache_profiles.module b/docroot/sites/all/modules/contrib/imagecache_profiles/imagecache_profiles.module
index 82d47ca..f4d4d53 100644
--- a/docroot/sites/all/modules/contrib/imagecache_profiles/imagecache_profiles.module
+++ b/docroot/sites/all/modules/contrib/imagecache_profiles/imagecache_profiles.module
@@ -35,7 +35,7 @@ function imagecache_profiles_preprocess_user_picture(&$variables) {
       $preset = variable_get('user_picture_imagecache_profiles_default', '');
     }
 
-    if (!empty($account->picture) && file_exists($account->picture)) {
+    if (!empty($account->picture) && (file_exists($account->picture) || module_exists('stage_file_proxy'))) {
       $picture = $account->picture;
     }
     elseif (variable_get('user_picture_default', '')) {
