diff --git a/uc_attribute/uc_attribute.module b/uc_attribute/uc_attribute.module
index 94e8311..fd63d64 100644
--- a/uc_attribute/uc_attribute.module
+++ b/uc_attribute/uc_attribute.module
@@ -1280,6 +1280,12 @@ function _uc_attribute_alter_form($id, &$product, $use_ajax) {
   $attributes = $product->attributes;
   $priced_attributes = uc_attribute_priced_attributes($nid);
 
+  // At this point, $product->data is the node author's userdata
+  // as a string, as populated by user_node_load(). We don't need that.
+  if (empty($product->data) || !is_array($product->data)) {
+    $product->data = array();
+  }
+
   // If the form is being built for the first time, populate attributes with their default values.
   if (!isset($product->data['attributes'])) {
     $values = array();
@@ -1302,10 +1308,6 @@ function _uc_attribute_alter_form($id, &$product, $use_ajax) {
     }
   }
 
-  if (empty($product->data) || !is_array($product->data)) {
-    $product->data = array();
-  }
-
   // Initialize the form element.
   $form_attributes = array(
     '#theme' => 'uc_attribute_add_to_cart',
