diff --git a/role_delay.module b/role_delay.module
index 923e418..691b537 100644
--- a/role_delay.module
+++ b/role_delay.module
@@ -135,8 +135,8 @@ function role_delay_cron() {
     foreach ($delays as $rid => $delay) {
       $results = db_query("SELECT u.name, r.* FROM {role_delay} r LEFT JOIN {users} u ON u.uid = r.uid WHERE waitperiod < :waitperiod AND r.rid = :rid", array(':waitperiod' => REQUEST_TIME, ':rid' => $rid));
       foreach ($results as $account) {
-        $id = db_insert('users_roles')
-          ->fields(array(
+        db_merge('user_roles')
+          ->key(array(
             'uid' => $account->uid,
             'rid' => $rid,
             ))
