diff --git a/role_delay.module b/role_delay.module
index 3fc110a..8f29555 100644
--- a/role_delay.module
+++ b/role_delay.module
@@ -119,11 +119,11 @@ function role_delay_user_presave(&$edit, $account, $category) {
 /**
  * Implements hook_user_delete().
  */
-function role_delay_user_delete($edit, $account, $method) {
+function role_delay_user_delete($account) {
   db_delete('role_delay')
-  ->condition('uid', $edit->uid)
+  ->condition('uid', $account->uid)
   ->execute();
-  watchdog('role_delay', 'Delayed roles for deleted user %name have been cleared.', array('%name' => $edit->name));
+  watchdog('role_delay', 'Delayed roles for deleted user %name have been cleared.', array('%name' => $account->name));
   }
 
 
