diff --git a/role_delay.module b/role_delay.module
index 3fc110a..60a5b7d 100644
--- a/role_delay.module
+++ b/role_delay.module
@@ -135,12 +135,18 @@ function role_delay_cron() {
     foreach ($delays as $rid => $delay) {
       $results = db_query("SELECT u.name, r.* FROM {role_delay} r LEFT JOIN {users} u ON u.uid = r.uid WHERE waitperiod < :waitperiod AND r.rid = :rid", array(':waitperiod' => REQUEST_TIME, ':rid' => $rid));
       foreach ($results as $account) {
-        $id = db_insert('users_roles')
-          ->fields(array(
-            'uid' => $account->uid,
-            'rid' => $rid,
+        $role = db_select('users_roles', 'r')
+          ->fields('r')
+          ->execute()
+          ->fetchAssoc();
+        if (!empty($role->rid)) {
+          db_insert('users_roles')
+            ->fields(array(
+              'uid' => $account->uid,
+              'rid' => $rid,
             ))
-          ->execute();
+            ->execute();
+        }
         db_delete('role_delay')
           ->condition('uid', $account->uid)
           ->condition('rid', $rid)
