diff --git a/sites/all/modules/patched/contact/contact.module b/sites/all/modules/patched/contact/contact.module
index 2ba4c38..e9c9285 100755
--- a/sites/all/modules/patched/contact/contact.module
+++ b/sites/all/modules/patched/contact/contact.module
@@ -136,6 +136,11 @@ function _contact_user_tab_access($account) {
     return FALSE;
   }
 
+  // Warmshowers specific: Members *are* allowed to contact other members
+  if (user_access('access user profiles')) {
+    return TRUE;
+  }
+
   // If the requested user has disabled their contact form, or this preference
   // has not yet been saved, do not allow users to contact them.
   if (empty($account->contact)) {
