diff --git a/uc_optional_checkout_review.install b/uc_optional_checkout_review.install
index d6a8f82..8e1b0c7 100644
--- a/uc_optional_checkout_review.install
+++ b/uc_optional_checkout_review.install
@@ -6,8 +6,28 @@
  */
 
 /**
- * Implementats of hook_uninstall().
+ * Implements hook_enable();
+ *
+ * The weight value selected here (99) must be higher than the weight for
+ * any other module that adds #submit handlers to uc_cart_checkout_form()
+ */
+function uc_optional_checkout_review_enable() {
+  db_update('system')
+    ->fields(array('weight' => 99))
+    ->condition('name', 'uc_optional_checkout_review', '=')
+    ->execute();
+}
+
+/**
+ * Set module weight so this module runs after any payment modules.
+ */
+function uc_optional_checkout_review_update_7101() {
+  uc_optional_checkout_review_enable();
+  return "Updated uc_optional_checkout_review weight in system table";
+}
+/**
+ * Implements hook_uninstall().
  */
 function uc_optional_checkout_review_uninstall() {
   variable_del('uc_optional_checkout_review');
-}
\ No newline at end of file
+}
diff --git a/uc_optional_checkout_review.module b/uc_optional_checkout_review.module
index f752b6d..33ea397 100644
--- a/uc_optional_checkout_review.module
+++ b/uc_optional_checkout_review.module
@@ -19,7 +19,10 @@ function uc_optional_checkout_review_form_uc_cart_checkout_settings_form_alter(&
 }
 
 /**
- * Implementats hook_form_FORM_ID_alter().
+ * Implements hook_form_FORM_ID_alter().
+ * This #submit handler must be *last* so that all others get their chance at
+ * processing.
+ * @see uc_optional_checkout_review_enable()
  */
 function uc_optional_checkout_review_form_uc_cart_checkout_form_alter(&$form, &$form_state, $form_id) {
   if (variable_get('uc_optional_checkout_review', FALSE)) {
@@ -32,7 +35,7 @@ function uc_optional_checkout_review_form_uc_cart_checkout_form_alter(&$form, &$
  * Submit handler
  */
 function uc_optional_checkout_review_uc_cart_checkout_form_submit($form, &$form_state) {
-  // NB: we explicitely check the form_id because of #2314427 (uc_payafter)
+  // NB: we explicitly check the form_id because of #2314427 (uc_payafter)
   if (variable_get('uc_optional_checkout_review', FALSE) && $form['#form_id'] == 'uc_cart_checkout_form') {
     $order = $form_state['storage']['order'];
 
