diff --git a/uc_optional_checkout_review.module b/uc_optional_checkout_review.module
index 90c5af5..f752b6d 100644
--- a/uc_optional_checkout_review.module
+++ b/uc_optional_checkout_review.module
@@ -34,7 +34,7 @@ function uc_optional_checkout_review_form_uc_cart_checkout_form_alter(&$form, &$
 function uc_optional_checkout_review_uc_cart_checkout_form_submit($form, &$form_state) {
   // NB: we explicitely check the form_id because of #2314427 (uc_payafter)
   if (variable_get('uc_optional_checkout_review', FALSE) && $form['#form_id'] == 'uc_cart_checkout_form') {
-    $order = uc_order_load($_SESSION['cart_order']);
+    $order = $form_state['storage']['order'];
 
     if ($order->payment_method == 'paypal_wps') {
       $wps_form = drupal_retrieve_form('uc_paypal_wps_form', $form_state, $order);
