diff --git a/views_json.module b/views_json.module
index c5230b5..4648926 100644
--- a/views_json.module
+++ b/views_json.module
@@ -153,8 +153,23 @@ function _views_json_render_fields($view, $row) {
       }
     }
     else {
+      // Get the formatted value of the field
       $field_output = $view->field[$field->options['id']]->advanced_render($row);
-      $field_raw = (isset($view->field[$id]->field_alias) && isset($row->{$view->field[$id]->field_alias})) ? $row->{$view->field[$id]->field_alias} : NULL;
+
+      // If a field, get the raw value of the field
+      $handler = $view->field[$id]->handler_type;
+      $fieldname = "{$handler}_$id";
+
+      // If the row has the field *as* a field, use it as the raw value
+      if ($handler == 'field' && isset($row->$fieldname)) {
+        $value_type = $view->field[$id]->options['group_column'];
+        $field_raw = isset($row->{$fieldname}[0]['raw'][$value_type]) ? $row->{$fieldname}[0]['raw'][$value_type] : "";
+      }
+      // Otherwise use the raw value that works for properties like title
+      else {
+        $field_raw = (isset($view->field[$id]->field_alias) && isset($row->{$view->field[$id]->field_alias})) ? $row->{$view->field[$id]->field_alias} : NULL;
+      }
+
       // Check if this is field is the entity's label.
       if (empty($field->base_table)) {
         $entity_info = entity_get_info($field->table);
