Index: persistent_login.module
===================================================================
--- persistent_login.module
+++ persistent_login.module
@@ -116,7 +116,17 @@
     // - E-mail verification is not required when a visitor creates an account.
     // - The form is not being executed by a user administrator.
     if (!variable_get('user_email_verification', 1) && variable_get('user_register', 1) == 1 && !user_access('administer users')) {
-      $alter_form = TRUE;
+      if (module_exists('logintoboggan')) {
+	    if (variable_get('logintoboggan_immediate_login_on_register' , 0)) {
+	      $alter_form = TRUE;
+	    }
+	    else{
+	      $alter_form = FALSE;
+	    }
+	  }
+	  else {
+	   $alter_form = TRUE;
+	  }
     }
   }
   if (!$alter_form) {
