diff --git a/role_delay.module b/role_delay.module
index e910c0e..b1a9d1a 100644
--- a/role_delay.module
+++ b/role_delay.module
@@ -94,17 +94,15 @@ function role_delay_user($op, &$edit, &$user, $category = NULL) {
       break;
 
     case 'update' :
-      if ($edit['status'] == 0) {
+      if (isset($edit['status']) && $edit['status'] === 0) {
         db_query('DELETE FROM {role_delay} WHERE uid = %d', $user->uid);
         watchdog('role_delay', 'Delayed roles for blocked user !name have been cleared.', array('!name' => l($user->name, 'user/' . $user->uid)));
-        drupal_set_message(t('Delayed roles for blocked user !name have been cleared.', array('!name' => l($user->name, 'user/' . $user->uid))));
       }
       break;
 
     case 'delete' :
       db_query('DELETE FROM {role_delay} WHERE uid = %d', $user->uid);
       watchdog('role_delay', 'Delayed roles for deleted user %name have been cleared.', array('%name' => $user->name));
-      drupal_set_message(t('Delayed roles for deleted user %name have been cleared.', array('%name' => $user->name)));
       break;
   }
 }
