diff -u -r ./includes/content.token.inc /Users/rfay/workspace/warmshowers/docroot/sites/all/modules/patched/cck/includes/content.token.inc
--- ./includes/content.token.inc	2011-01-05 03:34:57.000000000 -0700
+++ /Users/rfay/workspace/warmshowers/docroot/sites/all/modules/patched/cck/includes/content.token.inc	2015-08-31 15:55:59.000000000 -0600
@@ -166,6 +166,7 @@
       $tokens['user reference']['link']  = t('Formatted HTML link to referenced user');
       $tokens['user reference']['path']  = t("Relative path alias to the referenced user.");
       $tokens['user reference']['url']  = t("Absolute path alias to the referenced user.");
+      $tokens['user reference']['mail'] = t('Referenced user email address');
 
       return $tokens;
     }
@@ -181,6 +182,9 @@
       $tokens['path'] = is_numeric($item['uid']) ? url('user/' . $item['uid']) : '';
       $tokens['url'] = is_numeric($item['uid']) ? url('user/' . $item['uid'], array('absolute' => TRUE)) : '';
 
+      $user = user_load(array('uid' => $item['uid']));
+      $tokens['mail'] = $user ? $user->mail : '';
+
       return $tokens;
     }
   }
