diff --git a/lang_dropdown.module b/lang_dropdown.module
index 34cdb61..bb59f50 100644
--- a/lang_dropdown.module
+++ b/lang_dropdown.module
@@ -386,10 +386,10 @@ function _lang_dropdown_validate_output_value($element, &$form_state, $form) {
           form_error($element, t('You can\'t use !chosen output. You don\'t have !chosen library installed.', array('!chosen' => l(t('Chosen'), CHOSEN_WEB_URL))));
         }
         if (!module_exists('jquery_update')) {
-          form_error($element, t('Chosen needs jquery 1.7.2 or above. Install !jqueryupdate module to upgrade your drupal jquery version.', array('!jqueryupdate' => l(t('jquery update'), JQUERYUPDATE_MOD_URL))));
+          form_error($element, t('Chosen needs jquery 1.7 or above. Install !jqueryupdate module to upgrade your drupal jquery version.', array('!jqueryupdate' => l(t('jquery update'), JQUERYUPDATE_MOD_URL))));
         }
-        if (module_exists('jquery_update') && ((float) variable_get('jquery_update_jquery_version', '1.5') < 1.7)) {
-          form_error($element, t('Chosen needs jquery 1.7.2 or above. Configure !jqueryupdate module with a higher version.', array('!jqueryupdate' => l(t('jquery update'), 'admin/config/development/jquery_update'))));
+        if (module_exists('jquery_update') && (version_compare(variable_get('jquery_update_jquery_version', '1.5'), '1.7', '<'))) {
+          form_error($element, t('Chosen needs jquery 1.7 or above. Configure !jqueryupdate module with a higher version.', array('!jqueryupdate' => l(t('jquery update'), 'admin/config/development/jquery_update'))));
         }
       }
       break;
@@ -399,10 +399,10 @@ function _lang_dropdown_validate_output_value($element, &$form_state, $form) {
         form_error($element, t('You can\'t use !ddslick output. You don\'t have !ddslick library installed.', array('!ddslick' => l(t('ddSlick'), DDSLICK_WEB_URL))));
       }
       if (!module_exists('jquery_update')) {
-        form_error($element, t('ddSlick needs jquery 1.7.2 or above. Install !jqueryupdate module to upgrade your drupal jquery version.', array('!jqueryupdate' => l(t('jquery update'), JQUERYUPDATE_MOD_URL))));
+        form_error($element, t('ddSlick needs jquery 1.7 or above. Install !jqueryupdate module to upgrade your drupal jquery version.', array('!jqueryupdate' => l(t('jquery update'), JQUERYUPDATE_MOD_URL))));
       }
-      if (module_exists('jquery_update') && ((float) variable_get('jquery_update_jquery_version', '1.5') < 1.7)) {
-        form_error($element, t('ddSlick needs jquery 1.7.2 or above. Configure !jqueryupdate module with a higher version.', array('!jqueryupdate' => l(t('jquery update'), 'admin/config/development/jquery_update'))));
+      if (module_exists('jquery_update') && (version_compare(variable_get('jquery_update_jquery_version', '1.5'), '1.7', '<'))) {
+        form_error($element, t('ddSlick needs jquery 1.7 or above. Configure !jqueryupdate module with a higher version.', array('!jqueryupdate' => l(t('jquery update'), 'admin/config/development/jquery_update'))));
       }
       break;
     
