diff --git a/smtp.module b/smtp.module
index 48d86d4..4f6debc 100644
--- a/smtp.module
+++ b/smtp.module
@@ -463,7 +463,7 @@ function smtp_drupal_mail_wrapper($message) {
       case 'content-type':
   // Parse several values on the Content-type header, storing them in an array like
   // key=value -> $vars['key']='value'
-  $vars = explode('; ', $value);
+  $vars = preg_split('/;[\s]*/', $value);
   foreach ($vars as $i => $var) {
     if ($cut = strpos($var, '=')) {
       $new_var = drupal_strtolower(drupal_substr($var, $cut + 1));
