diff --git a/user_reference/user_reference.module b/user_reference/user_reference.module
index b2d6c8f..02d91b9 100644
--- a/user_reference/user_reference.module
+++ b/user_reference/user_reference.module
@@ -844,7 +844,12 @@ function _user_reference_potential_references_views($field, $options) {
 function _user_reference_potential_references_standard($field, $options) {
   // Avoid useless work.
   $filter_roles = array_filter($field['settings']['referenceable_roles']);
-  $filter_status = array_filter($field['settings']['referenceable_status']);
+  // $field['settings']['referenceable_status'] may be an int/boolean on D6 upgraded sites
+  $filter_status = array();
+  if (is_array($field['settings']['referenceable_status'])) {
+    // Selects only items in array that are not empty (true, 1, string, etc.)
+    $filter_status = array_filter($field['settings']['referenceable_status']);
+  }
   if (!count($filter_status) && !count($filter_roles)) {
     return array();
   }
